test/farm/001-farm.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+#
+# Tests of podman farm commands
+#
+
+load helpers.bash
+
+###############################################################################
+# BEGIN tests
+
+fname="test-farm"
+containerfile="test/farm/Containerfile"
+
+@test "farm - check farm has been created" {
+    run_podman farm ls
+    assert "$status" -eq 0 "get list of farms"
+    assert "$output" =~ $fname
+    assert "$output" =~ "test-node"
+}
+
+@test "farm - build on local only" {
+    iname="test-image-1"
+    empty_farm="empty-farm"
+    # create an empty farm
+    run_podman farm create $empty_farm
+    assert "$status" -eq 0 "create empty farm"
+    run_podman farm --farm $empty_farm build -f $containerfile -t $iname .
+    assert "$status" -eq 0 "create new manifest list $iname"
+    assert "$output" =~ "Local builder ready"
+    # inspect manifest list built and saved in dir
+    run_podman manifest inspect $iname
+    assert "$status" -eq 0 "inspect manifest list $iname"
+    assert "$output" =~ "amd64"
+    # cleanup images on node
+    ssh $ROOTLESS_USER@localhost podman rmi -af
+    assert "$status" -eq 0 "remove built image on remote node"
+}
+
+@test "farm - build on farm node and local" {
+    iname="test-image-2"
+    run_podman farm build -f $containerfile -t $iname .
+    assert "$status" -eq 0 "create new manifest list $iname"
+    assert "$output" =~ "Farm \"$fname\" ready"
+    # inspect manifest list built and saved in dir
+    run_podman manifest inspect $iname
+    assert "$status" -eq 0 "inspect manifest list $iname"
+    assert "$output" =~ "amd64"
+    # cleanup images on node
+    ssh $ROOTLESS_USER@localhost podman rmi -af
+    assert "$status" -eq 0 "remove built image on remote node"
+}
+
+@test "farm - build on farm node only with --cleanup" {
+    iname="test-image-3"
+    run_podman farm build -f $containerfile --cleanup --local=false -t $iname .
+    assert "$status" -eq 0 "create new manifest list $iname"
+    assert "$output" =~ "Farm \"$fname\" ready"
+    # inspect manifest list built and saved in dir
+    cat $iname/manifest.json
+    assert "$status" -eq 0 "inspect manifest list $iname"
+    assert "$output" =~ "amd64"
+    # see if we can ssh into node to check the image was cleaned up
+    ssh $ROOTLESS_USER@localhost podman images --filter dangling=true --noheading | wc -l
```

Comment:
Many problems with this, see above re: ssh, `ROOTLESS_USER`, PATH/podman, ...

Reply:
I need to check the output of podman images on the remote machine so have redirected the output of ssh to a variable and I check that variable for the content. Is this correct?

Reply:
I guess your current approach `foo=$(ssh ...)` will work, but I'd recommend two changes:
1. add `2>&1` to capture warnings/errors from ssh or podman or anything
2. add some sort of test for something that is not `""`, like a `podman images -q --short` and check for a SHA. To make sure that something happened.

Reply:
It does a cleanup of the images, so the test is checking for nothing being returned which would mean the image was removed successfully after the build was successful, hence the check with "".
Added `2>&1` to capture warnings/errors